src/main/java/com/iota/iri/network/Node.java

```diff
@@ -339,28 +348,42 @@ public void preProcessReceivedData(byte[] receivedData, SocketAddress senderAddr
     public void addReceivedDataToReceiveQueue(TransactionViewModel receivedTransactionViewModel, Neighbor neighbor) {
```

Comment:
* This comment refers to the methods addReceivedDataToReplyQueue, addReceivedDataToReceiveQueue and broadcast below.

As you can see in those methods, I changed the method that removes a message from the list from "poolLast()" to "pool()", due to differences in how the collections are implemented.
I'm aware that the new code below is not the same thing as the current one. I'll try to explain why I think it's still valid.

* Context for those who don't have the full picture of what this code does:
The Reply, Broadcast and Received message queues have a default limit size. They also have each one a Comparator, which is created by the time the queue is instanciated. That comparator makes the queue ordered, based on what aspect of the message is most relevant for each queue. (For more details, see the individual implementation of each Comparator). It suffices to say that each queue will have an order.
In the original code, in the unlikely event of the queue be full (i.e size > default limit), the new message would be added anyway, but the last one would be removed (remember, the "last" after the reordering).
In the new code, it also adds the message, but instead of the last, it removes the first one (method poll of the class PriorityBlockingQueue)

Now, in my opnion it doesn't make difference of which element we remove from the queue, first or last, because there's no way we be sure which element will be removed, because the comparator will always "reorder" the queue on each insert. For example, the removed element could be the one just added, in both cases, original and new implementation.

Also, ideally this code should be rarely executed. If the queues are getting full it means that the overall system is slow for some reason. This is just a fail safe to allow the system continue to run and not cause a OutOfMemory exception. I also added some warning logs, so the maintainer can be aware that messages are being dropped.